lib/spack/spack/main.py

```diff
@@ -962,7 +963,28 @@ def _main(argv=None):
 
     # Try to load the particular command the caller asked for.
     cmd_name = args.command[0]
-    cmd_name = aliases.get(cmd_name, cmd_name)
+
+    if cmd_name not in spack.cmd.all_commands():
+        aliases = spack.config.get("config:aliases")
+        alias = None
+
+        if aliases:
+            alias = aliases.get(cmd_name)
+
+        if alias is not None:
+            alias_parts = shlex.split(alias)
+            cmd_name = alias_parts[0]
+            args.command = alias_parts + args.command[1:]
+        else:
+            # If cmd_name is not an alias but also not a proper command,
```

Comment:
This seems new to me, and missing from the docs.  Am I understanding this that is like an under the hood autocomplete?

`spack ins zlib` would automatically get mapped to `spack install zlib`

Or is it integral to how the aliases work and I'm just missing it?

Reply:
This part was motivated/requested by @adamjstewart above. It's a separate commit and can be dropped easily if we don't want to include that functionality.

Reply:
I have mixed feelings about it.  I like it and I don't.  Encourages shortcuts which I love, could be confusing if you don't understand the metrics being used. 

My vote would be drop it so people have to explicitly define their aliases, but if we keep it I think it needs to be in the docs so we have an easy way to reference/explain the behavior when there are questions in the future.

Reply:
My vote is to keep it obviously 😄 

It's not confusing if you don't know it exists and don't use it.

Reply:
Sure. I just think the case to _not know it exists and accidentally use it_ is high.  We're psuedo supporting typos.  If there is a strong desire to keep then I just want us to be explicit that it is there in our documentation. I mean, tbh, I would probably use it now that I know it is there. I just don't look forward to explaining it to a bunch of people who ask me "why does spack do this?" 

Enough people in my sphere of influence already consider spack to be the great and mysterious Oz who does things the way it wants.  I don't want to accidentally reinforce that sterotype for power user and core dev conveniences. 😅

Reply:
Actually, I would fully support this if it were enabled by another config flag. I think a `power_user_mode` config option would be a nice umbrella flag for these sorts of things.  maybe separate PR though?

Reply:
Any thoughts @becker33 or @tgamblin? seems like a discussion you two would want to weigh in on.

Reply:
I would be fine with a power_user_mode flag but personally don't think it's necessary. Curious what others think though.

Reply:
I also don't think we need a power user mode. Other tools like `ip` also have this behavior. However, as I've mentioned, it's a separate commit and I would rather remove it before adding a power user mode or holding this PR any longer.

Reply:
@tgamblin can we sneak this into v0.21?

Reply:
@becker33 @tgamblin??? I would love that @adamjstewart. Not sure if there is the bandwidth. Let's see. 🤞

Reply:
Also, tab completion somewhat negates the need for the feature I requested so don't worry about me.